esp-wifi/src/compat/timer_compat.rs

```diff
@@ -83,12 +83,18 @@ pub fn compat_timer_disarm(ets_timer: *mut ets_timer) {
 
 pub fn compat_timer_done(ets_timer: *mut ets_timer) {
     critical_section::with(|_| unsafe {
-        if let Some(timer) = TIMERS.iter_mut().find(|t| t.ets_timer == ets_timer) {
+        if let Some((idx, timer)) = TIMERS
+            .iter_mut()
+            .enumerate()
+            .find(|(_, t)| t.ets_timer == ets_timer)
+        {
             debug!("timer_done {:x}", timer.id());
             timer.active = false;
 
             (*ets_timer).priv_ = core::ptr::null_mut();
             (*ets_timer).expire = 0;
+
+            TIMERS.swap_remove(idx);
```

Comment:
I think we also need to free the `ets_timer_t` we allocate in `compat_esp_timer_create`?

Reply:
oh true